commons/src/main/java/fr/free/nrw/commons/upload/ImageProcessing.java

```diff
@@ -0,0 +1,127 @@
+package fr.free.nrw.commons.upload;
+
+import android.content.Context;
+import android.database.Cursor;
+import android.media.ExifInterface;
+import android.net.Uri;
+import android.provider.DocumentsContract;
+import android.provider.MediaStore;
+import android.util.Log;
+
+import java.io.IOException;
+
+
+/**
+ * Created by misao on 16-Dec-15.
+ */
+public class ImageProcessing {
+
+    private Uri uri;
+    private Context context;
+
+    public ImageProcessing(Context context, Uri uri){
+        this.context = context;
+        this.uri = uri;
+    }
+
+    //Gets file path of image from its Uri
```

Comment:
Sorry, a bit confused: What sort of situation would cause it to return null, as the method is only called when the user selects an image for uploading? If he can select it, that means the image and thus the Uri and file path exist right?